src/main/js/comment/CreateComment.tsx

```diff
@@ -189,18 +196,54 @@ class CreateComment extends React.Component<Props, State> {
             <div className="media-content">
               <div className="field">
                 <div className="control">
-                  <Textarea
-                    name="comment"
-                    value={this.state.newComment.comment}
-                    autofocus={autofocus}
+                  <MentionTextarea
+                    value={newComment?.comment}
+                    onChange={this.handleChanges}
+                    onSubmit={this.submit}
                     placeholder={t(
-                      this.state.newComment.type === "TASK_TODO"
+                      newComment?.type === "TASK_TODO"
                         ? "scm-review-plugin.comment.addTask"
                         : "scm-review-plugin.comment.addComment"
                     )}
-                    onChange={this.handleChanges}
-                    onSubmit={this.submit}
-                  />
+                  >
+                    <Mention
+                      markup="@[__id__]"
+                      displayTransform={(id: string) => {
+                        const { newComment } = this.state;
+                        return newComment?.mentions && newComment.mentions.length > 0
+                          ? `@${newComment.mentions?.filter(entry => entry.id === id)[0]?.displayName}`
+                          : `@${id}`;
+                      }}
+                      trigger="@"
+                      data={(query, callback) =>
+                        mapAutocompleteToSuggestions(this.props.userAutocompleteLink, query, callback)
+                      }
+                      onAdd={(id, displayName) => {
+                        this.setState(prevState => ({
+                          ...prevState,
+                          newComment: {
+                            ...prevState.newComment,
+                            mentions: [...prevState.newComment.mentions, { id, displayName }]
+                          }
+                        }));
+                      }}
+                      renderSuggestion={(
+                        suggestion: SuggestionDataItem,
+                        search: string,
+                        highlightedDisplay: React.ReactNode,
+                        index: number,
+                        focused: boolean
+                      ) => (
+                        <StyledSuggestion className="user" focused={focused} index={index}>
+                          {highlightedDisplay}
+                        </StyledSuggestion>
+                      )}
+                      style={{
+                        color: "transparent"
+                      }}
+                      appendSpaceOnAdd={true}
+                    />
```

Comment:
This whole block should be moved to the MentionTextarea. You just need one or two more props and a callback method.